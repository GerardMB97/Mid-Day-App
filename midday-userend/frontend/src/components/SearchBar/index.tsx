import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import { TextInput, View, StyleSheet } from 'react-native';
import colors from '../../../colors';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    width: '100%',
    paddingLeft: 20,
    paddingRight: 20
  },
  input: {
    width: '100%',
    height: 60,
    backgroundColor: colors.green,
    justifyContent: 'center',
    paddingLeft: 40,
    marginBottom: 20,
    borderRadius: 10
  },
  search: {
    fontSize: 25,
    color: 'black',
    position: 'absolute',
    left: '10%',
    top: '20%'
  }
});

export default function SearchBar ({ inputPlaceholder, setInputValue, inputValue, action }: {inputPlaceholder: string, setInputValue:any, inputValue:string, action:any}) {
  return (
    <View style = {styles.container}>
     <TextInput
      style = {styles.input}
      placeholder={inputPlaceholder}
      value={inputValue}
      onChangeText={(text) => { setInputValue(text); action(text); }}>
      </TextInput>
      <Icon style={styles.search} name= "search-outline"></Icon>
      </View>);
};
