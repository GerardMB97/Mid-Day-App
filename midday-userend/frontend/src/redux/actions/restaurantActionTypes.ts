const restaurantActionTypes = {
  LOAD_CATEGORIES: 'LOAD_CATEGORIES',
  FILTER_CATEGORIES: 'FILTER_CATEGORIES',
  LOAD_RESTAURANTS: 'LOAD_RESTAURANTS'
};

export default restaurantActionTypes
;
