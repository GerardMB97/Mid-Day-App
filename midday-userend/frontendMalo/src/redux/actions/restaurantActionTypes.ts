const restaurantActionTypes = {
  LOAD_CATEGORIES: 'LOAD_CATEGORIES'
};

export default restaurantActionTypes
;
