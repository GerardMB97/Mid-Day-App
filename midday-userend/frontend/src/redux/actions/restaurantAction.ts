import axios from 'axios';
import restaurantActionTypes from './restaurantActionTypes';

export const loadCategories = () => {
  return async (dispatch: Function) => {
    const categories = await axios.get('http://localhost:5000/api/midday/restaurants/categories');

    dispatch({
      type: restaurantActionTypes.LOAD_CATEGORIES,
      categories
    });
  };
}
;
