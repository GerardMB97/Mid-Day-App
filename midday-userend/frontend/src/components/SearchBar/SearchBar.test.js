import React from 'react';
import SearchBar from '.';
import { fireEvent, render } from 'react-native-testing-library';
import { filterCategories } from '../../redux/actions/restaurantAction';

jest.mock('../../redux/actions/restaurantAction');
jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');

describe('Given a component SearchBar', () => {
  test('Then it should render correctly', () => {
    const rendered = render(
      <SearchBar
      inputPlaceholder ='hello'
      setInputValue = {jest.fn()}
      inputValue = ''
      action = {jest.fn()}
      ></SearchBar>
    );
    expect(rendered).toMatchSnapshot();
  });
  describe('When input Value changes', () => {
    test('Then action should be invoked', () => {
      const tree = render(
       <SearchBar
      inputPlaceholder ='hello'
      setInputValue = {() => {}}
      inputValue = ''
      action = {filterCategories}
      ></SearchBar>
      );

      const textInput = tree.getByPlaceholder('hello');
      fireEvent.changeText(textInput, 'asiatica');

      expect(filterCategories).toHaveBeenCalled();
    });
  });
});
