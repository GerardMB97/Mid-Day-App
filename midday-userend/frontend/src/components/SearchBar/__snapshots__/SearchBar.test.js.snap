// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a component SearchBar Then it should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "paddingLeft": 20,
      "paddingRight": 20,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="hello"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "rgb(135, 175, 147)",
        "borderRadius": 10,
        "height": 60,
        "justifyContent": "center",
        "marginBottom": 20,
        "paddingLeft": 40,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Icon
    name="search-outline"
    style={
      Object {
        "color": "black",
        "fontSize": 25,
        "left": "10%",
        "position": "absolute",
        "top": "20%",
      }
    }
  />
</View>
`;
